File: src/main/java/fm/comunas/ddd/part/model/Part.java
package fm.comunas.ddd.part.model;

import fm.comunas.ddd.aggregate.model.Aggregate;

/**
 * A Part is an Entity that belongs to an Aggregate.
 */
public interface Part<A extends Aggregate> {

	PartMeta<A> getMeta();

	Integer getId();

	Integer getParentPartId();

	String getPartListTypeId();

	Integer getSeqNr();

	void setSeqNr(Integer seqNr);

}

File: fm-dddrive/src/main/java/io/dddrive/ddd/model/Part.kt
package io.dddrive.ddd.model

import io.dddrive.property.model.EntityWithProperties

/**
 * A Part is an Entity that belongs to an Aggregate (but might be attached to another part as parent).
 */
interface Part<A : Aggregate> : EntityWithProperties {

	val id: Int

	val meta: PartMeta<A>

}
